update-content-templates/action.yaml

```diff
@@ -0,0 +1,21 @@
+name: Update Content Templates
+description: Update Content Templates Configuration. [Documentation](https://github.com/zingdevlimited/actions-helpers/blob/v3/docs/usage/composite-actions-general.md#update-content-templates)
+
+inputs:
```

Comment:
Consider adding an optional flag **ALWAYS_OVERWRITE**, where if it is equal `true` then the code checks if the existing template content is different from what's in the configuration file.

If so, can it Delete and then Recreate the content template with the new value (to force an update)? Unless that doesn't work?

Reply:
I think an ALWAYS_OVERWRITE flag is a good idea, however I wonder if for the first version it should always overwrite, without checking if the existing template content and configuration file are different. This then provides the ability to replace existing templates if required due to updated config, and then in a later version, it can evaluate the configuration of the existing template with the new config to see if it is different. What do you think?

Reply:
The point of the difference check is to avoid over-calling the API by deleting and creating every single time even when nothing has changed.
I'm not sure what you mean as the point is that you have 3 scenarios:

- template does not exist. create new
- template exists but has different content (just do a basic string compare). Delete it and create a new one
- template exists but has the same content. Skip it like it does right now

Reply:
It was my understanding that the comparison would be more complex which is why I suggested adapting it in the future, but if it is simply a string compare then should be fine. To do it as a string compare, would we just stringify the template from the response and from the config and compare them?

Reply:
I'd go a level down. The `types` object and `variables` object?. Stringify with no whitespace. If both sides start as Objects and then are stringified by the same stringifier then it should provide the keys with the same order

![image](https://github.com/user-attachments/assets/758b0563-2f61-47cd-bc19-cecd76ad159e)